TestRunner/TestRunnerOperation.swift

```diff
@@ -154,10 +156,26 @@ class TestRunnerOperation: Operation {
         guard !simulatorLaunched else { return }
         
         simulatorLaunched = true
-        
+        connectExternalDisplayToSimulator()
```

Comment:
Instead of passing `needsExternalDisplay` through everywhere, all you really need to do is wrap this function with this:
```
if AppArgs.shared.needsExternalDisplay {
  connectExternalDisplayToSimulator()
}
```
Then remove the guard from within the function. You shouldn't need all the other variables and other stuff you added.